docs/source/en/tasks/prompting.md

```diff
@@ -0,0 +1,466 @@
+[`tiiuae/falcon-7b`](https://huggingface.co/tiiuae/falcon-7b) and [`tiiuae/falcon-7b-instruct`](https://huggingface.co/tiiuae/falcon-7b-instruct).
+
+Base models are excellent at completing the text when given an initial prompt, however, they are not ideal for NLP tasks 
+where they need to follow instructions, or for conversational use. This is where the instruct (chat) versions come in. 
+These checkpoints are the result of further fine-tuning of the pre-trained base versions on instructions and conversational data. 
+This additional fine-tuning makes them a better choice for many NLP tasks.  
+
+Let's illustrate some simple prompts that you can use with [`tiiuae/falcon-7b-instruct`](https://huggingface.co/tiiuae/falcon-7b-instruct) 
+to solve some common NLP tasks.
+
+### NLP tasks 
+
+First, let's set up the environment: 
+
+```bash
+pip install -q transformers einops accelerate
+```
+
+Next, let's load the model with the appropriate pipeline (`"text-generation"`): 
+
+```python
+>>> from transformers import pipeline, AutoTokenizer
+>>> import torch
+
+>>> torch.manual_seed(0) # doctest: +IGNORE_RESULT
+>>> model = "tiiuae/falcon-7b-instruct"
+
+>>> tokenizer = AutoTokenizer.from_pretrained(model)
+>>> pipe = pipeline(
+...     "text-generation",
+...     model=model,
+...     tokenizer=tokenizer,
+...     torch_dtype=torch.bfloat16,
+...     trust_remote_code=True,
+...     device_map="auto",
+... )
+```
+
+<Tip>
+
+Note that Falcon models were trained using the `bfloat16` datatype, so we recommend you use the same. This requires a recent 
+version of CUDA and works best on modern cards.
+
+You also need to allow remote code execution. This is because the Falcon models use a new architecture that is not part of transformers yet - instead, the code necessary is provided by the model authors in the repo.
+
+</Tip>
+
+Now that we have the model loaded via the pipeline, let's explore how you can use prompts to solve NLP tasks.
+
+#### Text classification
+
+One of the most common forms of text classification is sentiment analysis, which assigns a label like "positive", "negative", 
+or "neutral" to a sequence of text. Let's write a prompt that instructs the model to classify a given text (a movie review). 
+We'll start by giving the instruction, and then specifying the text to classify. Note that instead of leaving it at that, we're 
+also adding the beginning of the response - `"Sentiment: "`:
+
+```python
+>>> torch.manual_seed(0) # doctest: +IGNORE_RESULT
+>>> prompt = """Classify the text into neutral, negative or positive. 
+... Text: This movie is definitely one of my favorite movies of its kind. The interaction between respectable and morally strong characters is an ode to chivalry and the honor code amongst thieves and policemen.
+... Sentiment:
+... """
+
+>>> sequences = pipe(
+...     prompt,
+...     max_length=100,
```

Comment:
```suggestion
...     max_new_tokens=10,
```
Think it's better to use `max_new_tokens=10` instead of `max_length` here cc @gante